app/services/procurer/service.rb

```diff
@@ -0,0 +1,80 @@
+module Procurer
+  class Service
+    def initialize(name)
+      @name = name
+    end
+
+    # delegate the call the subclass
+    delegate :call, to: :instance_klass
+
+    def instance_klass
+      debugger
+      klass = "Procurer::#{@name.capitalize}".constantize
+      klass.new
+    rescue NameError
+      debugger
```

Comment:
Ah the debugger should be removed outside of the implementation here? (besides local dev I mean)

Reply:
whoops. good spot! ill remove it 🙏